### ISSUE TITLE
Double clicking on the drag bar to maximize does not update the maximize button icon

### ISSUE THREAD

**Issue by ocalvo** [microsoft/terminal#1780]
Double clicking on the drag bar to maximize does not update the maximize button icon, the icon remains the same, by contrast, if you click the maximize button the icon does update.

# Environment

```none
Windows 10 19H1
Windows Terminal version (if applicable): master

Any other software?
```

# Steps to reproduce
1. Start Windows Terminal in normal state
2. Double click the drag bar to maximize the window.

# Expected behavior
The maximize button icon is updated

# Actual behavior
The maximize button icon does not update.

We are not listening from window state change events from the top level window into the mixmaxclose user control.

**Comment by gopalrander** [comment]
Debugged and noticed that MinMaxCloseControl::DragBar_DoubleTapped() method is not hit when I double clicked the Drag bar. [But the window maximizes; strange?

**Comment by DHowett-MSFT** [comment]
@gopalrander the drag bar is actually cut out of the window using a region. It's a complicated story about _why exactly_ we need to do this.

**Comment by shanselman** [comment]
Came here to make this bug. ;) Hitting it daily now. Can't we grab the event and delegate it over?

**Comment by filip-hejsek** [comment]
Maximizing the window by dragging it to the top of the screen has the same problem

**Comment by nishp77** [comment]
@filip-hejsek do u knw abt xaml ?? and uwp ??

**Comment by zadjii-msft** [comment]
If anyone's interested in fixing this, I think I broke this in #2438.

The trick is that I think we're now treating the titlebar area as `HTCAPTION` in NonClientIslandWindow, so the mouse presses aren't falling through to the underlying Xaml Island window (the one that draws the TitlebarControl).

We should be able to handle the double-click directly in NonClientIslandWindow directly. I'm a bit curious of _why_ it regressed, but as long as a fix doesn't regress #2028, I'm happy.

**Comment by DHowett-MSFT** [comment]
Nah, it never was hitting the island here. Double click on a Win32 titlebar always did a maximize by default :)
